test/wrapper_test.rb

```diff
@@ -3040,8 +3046,28 @@ def test_assert_inapplicable_relations
     }]
 
     vrp = TestHelper.create(problem)
-    assert_includes OptimizerWrapper.config[:services][:vroom].inapplicable_solve?(vrp), :assert_no_relations
-    refute_includes OptimizerWrapper.config[:services][:ortools].inapplicable_solve?(vrp), :assert_no_relations
+    assert_includes OptimizerWrapper.config[:services][:vroom].inapplicable_solve?(vrp),
+                    :assert_no_relations_except_simple_shipments
+    refute_includes OptimizerWrapper.config[:services][:ortools].inapplicable_solve?(vrp),
+                    :assert_no_relations_except_simple_shipments
+
+    problem = VRP.basic
+    problem[:relations] = [{
+      type: :shipment,
+      linked_ids: ['service_1', 'service_2']
+    }]
+    problem[:vehicles].first[:capacities] = [{ unit_id: 'kg', limit: 2 }]
+    problem[:services][0][:quantities] = [{ unit_id: 'kg', value: 1 }]
+    problem[:services][1][:quantities] = [{ unit_id: 'kg', value: -1 }]
+    vrp = TestHelper.create(problem)
+    refute_includes OptimizerWrapper.config[:services][:vroom].inapplicable_solve?(vrp),
+                    :assert_no_relations_except_simple_shipments
+
+    problem[:services][1][:quantities] = [{ unit_id: 'kg', value: 2 }]
+    vrp = TestHelper.create(problem)
+    assert_includes OptimizerWrapper.config[:services][:vroom].inapplicable_solve?(vrp),
```

Comment:
Shouldn't this relation be rejected by vrp.create ?

Reply:
OR-Tools can handle a shipment with a pickup of -1 and a delivery of -2. There no reason to refuse a problem with such quantities